### PR TITLE
GH-355 Made user_dn_pattern of artifactory_ldap_setting optional

### PR DESCRIPTION
**Pull request by maheshjfrog**
Fixes: #355

### CODE REVIEW

File: pkg/artifactory/resource_artifactory_ldap_setting.go [hunk 1]
```diff
@@ -68,9 +68,9 @@ func resourceArtifactoryLdapSetting() *schema.Resource {
 		},
 		"user_dn_pattern": {
 			Type:             schema.TypeString,
-			Required:         true,
+			Optional:         true,
 			ValidateDiagFunc: validation.ToDiagFunc(validation.All(validation.StringIsNotEmpty, validateLdapDn)),
```
Comment (chb0github) [inline pkg/artifactory/resource_artifactory_ldap_setting.go:72]: Does allowing the key but effectively no value make sense?? I guess. If you're going to supply one, it has to be valid.

Please confirm with the user that IF they supply a user dn, then it must be valid

File: pkg/artifactory/resource_artifactory_ldap_setting_test.go [hunk 2]
```diff
@@ -139,6 +139,77 @@ resource "artifactory_ldap_setting" "ldaptestemailattr" {
 	})
 }
 
+func TestAccLdapSetting_user_dn_or_search_filter(t *testing.T) {
```
Comment (chb0github) [inline pkg/artifactory/resource_artifactory_ldap_setting_test.go:142]: Good work on adding tests

**Review by chb0github — APPROVED** [review]
Please confirm with the user that IF they supply a DN, then it must be valid.

File: pkg/artifactory/resource_artifactory_ldap_setting.go (hunk 1 above)
Comment (maheshjfrog) [reply pkg/artifactory/resource_artifactory_ldap_setting.go:72]: Sure. Will check with user about this.

**Comment by maheshjfrog** [comment]
Notes on the fix:
From UI, Artifactory"user_dn_pattern" allows blank input for this "user_dn_pattern" field if the "search_filter" field is specified.
Provider uses "system/configuration" REST API call which allows both the fields to be blank and the call does not validate each field of the ldap config like in UI.
Currently our provider can not enforce rule ie: either "user_dn_pattern" or "search_filter" has to be provided / both can not be empty.

**Review by chb0github — APPROVED** [review]

**Pull request merged**